Automattic/wp-calypso
# Gravatar: Magic login improvements

Related to 105749-Automattic/gravatar

## Proposed Changes

* Align the magic login page with the new design (IXT40Xck14laWK456R4wB2-fi)
* Bring the link of the password/social login page back
* Customize the links and the submit button of the WPJM login page
* Show the "Resend the verification email" function when `resendEmailCountdown` is `0` 
* Add track events:
  * `calypso_${ oauth2Client.name }_magic_login`
  * `calypso_${ oauth2Client.name }_magic_login_click_login_page_link`
  * `calypso_${ oauth2Client.name }_magic_login_click_resend_email`
  * `calypso_${ oauth2Client.name }_magic_login_click_use_different_email`

## Testing Instructions



* Check out this PR and run it
* Logout your account
* Go to the magic login page of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), and you will the design have been updated:

> Note: The `a` link of the "a WordPress.com account" will be implemented once it's ready.

| Gravatar | WPJM |
| - | - |
| <img width="853" alt="截圖 2023-09-15 下午3 57 08" src="https://github.com/Automattic/wp-calypso/assets/21308003/b324811b-5e01-4b16-9218-751cbe838771"> | <img width="853" alt="截圖 2023-09-15 下午3 56 54" src="https://github.com/Automattic/wp-calypso/assets/21308003/1f5e76d6-c4e3-442c-8e6d-3df7811d6d8f"> |

* Go to the password/social login page of [Gravatar](http://calypso.localhost:3000/log-in?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), and you will the color of the links and submit button have been updated:

| Gravatar | WPJM |
| - | - |
| <img width="853" alt="截圖 2023-09-15 下午3 57 26" src="https://github.com/Automattic/wp-calypso/assets/21308003/a3775c37-4667-4ea6-8755-c49dc51a5e32"> | <img width="853" alt="截圖 2023-09-15 下午3 56 19" src="https://github.com/Automattic/wp-calypso/assets/21308003/3fcb3252-0285-40f8-80f6-caea593b3f8e"> |

* Login via the magic login of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), and wait for 90 seconds, you will see the "Resend verification email" has been shown and it should work as expected:

https://github.com/Automattic/wp-calypso/assets/21308003/6e323902-940c-4967-a29d-51e7b3885de2

* Go to the magic login page of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), and check these events:

**When landing on the magic login page:**
<img width="971" alt="截圖 2023-09-15 下午4 14 43" src="https://github.com/Automattic/wp-calypso/assets/21308003/160db443-a82a-46fe-8351-f9087fd363f2">

**When clicking the "Sign in another way" link:**
<img width="966" alt="截圖 2023-09-15 下午4 44 13" src="https://github.com/Automattic/wp-calypso/assets/21308003/bfb3c221-930e-40a0-8ffb-78b247e7006d">

**When clicking the "Resend verification email" link:**
<img width="970" alt="截圖 2023-09-15 下午4 46 19" src="https://github.com/Automattic/wp-calypso/assets/21308003/9a53026c-dd9c-4d11-a3cc-caeb58682b92">

**When clicking the "Use different email address" link:**
<img width="968" alt="截圖 2023-09-15 下午4 48 18" src="https://github.com/Automattic/wp-calypso/assets/21308003/8c14caec-242c-455e-93b9-c8ecd13d4a61">

* Go to the [password/social log page](http://calypso.localhost:3000/log-in) and [magic login](http://calypso.localhost:3000/log-in/link) page of WP.com, they should look the same.

## Pre-merge Checklist
